react/features/large-video/components/LargeVideo.web.js

```diff
@@ -126,6 +166,32 @@ class LargeVideo extends Component<Props> {
         );
     }
 
+    _updateLayout: () => void;
+
+    /**
+     * Refreshes the video layout to determine the dimensions of the stage view.
+     * If the filmstrip is toggled it adds CSS transition classes and removes them
+     * when the transition is done.
+     *
+     * @returns {void}
+     */
+    _updateLayout() {
+        const { _verticalFilmstripWidth, _resizableFilmstrip } = this.props;
+
+        if (_resizableFilmstrip && _verticalFilmstripWidth >= FILMSTRIP_BREAKPOINT) {
+            this._containerRef.current.classList.add('transition');
+            this._wrapperRef.current.classList.add('transition');
```

Comment:
Why don't we do the class addition in the render method? why do we need the reference at all?

Reply:
I tried but it doesn't work. For transitions to work the CSS transition property must be on the element before the modification. If we try to add it on render then the new width and the transition get applied at the same time so the transition is not visible